CSLabs.Api/Models/ModuleModels/Module.cs

```diff
@@ -56,6 +59,26 @@ public async Task SetUserModuleIdIfExists(DefaultContext context, User user)
                 UserModuleId = userModule.Id;
         }
 
+        public void AddModuleTags(DefaultContext context)
+        {
+            foreach (var moduleTag in ModuleTags.Where(moduleTag => moduleTag.TagId == 0))
+            {
+                context.Add(moduleTag);
+                if (moduleTag.Tag.Id != 0)
+                    context.Entry(moduleTag.Tag).State = EntityState.Unchanged;
+            }
+        }
+
+        public void DeleteModuleTags(DefaultContext context)
+        {
+            var currentModuleTags = context.Modules.AsNoTracking().IncludeTags().First(m => m.Id == Id).ModuleTags;
+            var tagsToBeDeleted = currentModuleTags.Where(mt => !ModuleTags.Contains(mt)).ToList();
```

Comment:
No need to override the equals check, that could be problematic if we are ever comparing instances somewhere else.
Just use the .Any method.
```suggestion
            var tagsToBeDeleted = currentModuleTags.Where(mt => !ModuleTags.Any(m => m.TagId == mt.TagId)).ToList();
```